plugin/vscode/performance-writer.js

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2020  Matthew Glazar
+// See end of file for extended copyright information.
+
+const fs = require('fs');
+const path = require("path");
+const process = require("process");
+
+class PerformanceWriter {
+    
+    constructor (extensionContext) {
+        this._performanceTxtPath = path.join(process.cwd(), "test_performance.txt");
+        this._dataAttributes = {
+            columns:'timestamp,file-ref,file-name-or-duration,event-id'
+        };
+        this._fileName;
+        this._fileIndex;
+        this._extensionContext = extensionContext;
+    }
+
+    fileNameOptimizer (){
+        let currentSessionMap = this._extensionContext.globalState.get('qljsFileMap');
+        this._fileName = this._fileName.split(path.sep);
+        this._fileName = this._fileName[this._fileName.length - 1];
+        if (!(currentSessionMap)) {
+            currentSessionMap = {};
+            currentSessionMap['latestIndex'] = 1;
+            currentSessionMap[this._fileName] = 1;
```

Comment:
Minor: What if the user opens a JavaScript file called `latestIndex`?

Reply:
The variable is now called *latestIndex because js file cannot be named with a *

Reply:
> The variable is now called *latestIndex because js file cannot be named with a *

It can on Linux!

Reply:
[Addressed in latest commit]

the file index of the current vscode session is now called this._fileIndex and no longer a key within a object